nomenklatura/index/duckdb_index.py

```diff
@@ -0,0 +1,162 @@
+from duckdb import DuckDBPyRelation
+from followthemoney.types import registry
+from pathlib import Path
+from shutil import rmtree
+from typing import Iterable, Tuple
+import csv
+import duckdb
+import logging
+
+from nomenklatura.dataset import DS
+from nomenklatura.entity import CE
+from nomenklatura.index.common import BaseIndex
+from nomenklatura.index.tokenizer import NAME_PART_FIELD, WORD_FIELD, Tokenizer
+from nomenklatura.resolver import Pair, Identifier
+from nomenklatura.store import View
+
+log = logging.getLogger(__name__)
+
+BATCH_SIZE = 1000
+
+
+class DuckDBIndex(BaseIndex[DS, CE]):
```

Comment:
doesn't implement match() yet. Should be easy enough but probably needs `frequencies` materialised to a table.